db2i_conversion.cc

```diff
@@ -638,24 +612,9 @@ int ha_ibmdb2i::getFieldTypeMapping(Field* field,
               {
                 fieldLength = min(MAX_BLOB_LENGTH, fieldLength);
 
-                if (fieldCharSet->mbmaxlen > 1)
+                if (fieldCharSet->mbmaxlen > 1 && memcmp(fieldCharSet->csname, "utf8", 4) != 0)
```

Comment:
Are we guaranteed that fieldCharSet->csname will have 4 bytes allocated? Should we go for `strcmp()` instead? (repeat comment for other occurrences)

Reply:
Good point, `gbk` is an example charset that has `mbmaxlen > 1` but is only `3` characters long. We should consider moving this a `strcmp` instead.

Reply:
Can't use `strcmp` as we're doing a substring check for "utf8", "utf8mb3", and "utf8mb4". We could use `strncmp`, though.